docs/tutorial/onchain-rln-relay-chat2.md

```diff
@@ -0,0 +1,242 @@
+
+```
+>> Hi
+--rln epoch: 165886530
+<Jul 26, 12:55> Alice: Hi
+```
+The numerical value `165886530` indicates the epoch of the message `Hi`.
+You will see a different value than `165886530` on your screen. 
+If two messages sent by the same chat2 client happen to have the same RLN epoch value, then one of them will be detected as spam and won't be routed (by test fleets in this test setting).
+At the time of this tutorial, the epoch duration is set to `10` seconds.
+You can inspect the current epoch value by checking the following [constant variable](https://github.com/status-im/nim-waku/blob/21cac6d491a6d995a7a8ba84c85fecc7817b3d8b/waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim#L119) in the nim-waku codebase.
+Thus, if you send two messages less than `10` seconds apart, they are likely to get the same `rln epoch` values.
+
+After sending a chat message, you may experience some delay before the next chat prompt appears. 
+The reason is that under the hood a zero-knowledge proof is being generated and attached to your message.
+
+
+Try to spam the network by violating the message rate limit i.e.,
+sending more than one message per epoch. 
+Your messages will be routed via test fleets that are running in spam-protected mode over the same content topic i.e., `/toy-chat/2/luzhou/proto` as your chat client.
+Your spam activity will be detected by them and your message will not reach the rest of the chat clients.
+You can check this by running a second chat user and verifying that spam messages are not displayed as they are filtered by the test fleets.
+A sample test scenario is illustrated in the [Sample test output section](#sample-test-output).
+
+Once you are done with the test, make sure you close all the chat2 clients by typing the `/exit` command.
+```
+>> /exit
+quitting...
+```
+
+## How to reuse RLN credential
+
+You may resue your old RLN credential using `rln-relay-membership-index`, `rln-relay-id` and `rln-relay-id-commitment` options. 
+For instance, if the previously generated credential are
+```
+your membership index is: 63
+your rln identity key is: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+your rln identity commitment key is: 6c6598126ba10d1b70100893b76d7f8d7343eeb8f5ecfd48371b421c5aa6f012
+```
+Then, the execution command will look like this (inspect the last three config options):
+```
+./build/chat2  --fleet:test --content-topic:/toy-chat/2/luzhou/proto --rln-relay:true --rln-relay-dynamic:true --eth-mem-contract-address:0x5DE1Fb10345Ef1647629Df30e6C397297Da09A1d  --eth-account-address:your_eth_account --eth-account-privatekey:your_eth_private_key  --eth-client-address:your_goerli_node  --ports-shift=1  --rln-relay-membership-index:63 --rln-relay-id:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx --rln-relay-id-commitment:6c6598126ba10d1b70100893b76d7f8d7343eeb8f5ecfd48371b421c5aa6f012
```

Comment:
Contract address is incorrect. Needs to be `0x4252105670fE33D2947e8EaD304969849E64f2a6`

Reply:
Thanks, fixed in 0feab5efc211dfd2570144c50e7282bba4d80237